### PR TITLE
Dotty

### PR DESCRIPTION
**Pull request by cquiroz**
This PR adds support for dotty. I'm also proposing to switch to `sbt-github-actions` which seem to be the standard these days

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -34,33 +34,51 @@ lazy val docs = project
...
 val kindProjectorV = "0.11.2"
-val betterMonadicForV = "0.3.1"
+
+val Scala212 = "2.12.12"
+val Scala213 = "2.13.4"
+val Scala3 = Seq("3.0.0-M2", "3.0.0-M3")
+
+ThisBuild / crossScalaVersions := Seq(Scala212, Scala213) ++ Scala3
+ThisBuild / scalaVersion := Scala213
+ThisBuild / organization := "org.typelevel"
```
Comment (cquiroz) [inline build.sbt:44]: I think this is correct, but please let me know if I should revert it to `chrisdavenport`

File: core/src/main/scala/io/chrisdavenport/monoids/Dual.scala [hunk 2]
```diff
@@ -3,7 +3,7 @@ package io.chrisdavenport.monoids
 import cats._
 import cats.implicits._
 
-final case class Dual[A](getDual: A) extends AnyVal
+final case class Dual[A](getDual: A)
```
Comment (cquiroz) [inline core/src/main/scala/io/chrisdavenport/monoids/Dual.scala:6]: Dotty seems to have troubles with generic classes extending `AnyVal` but once removed everything works

Comment (ChristopherDavenport) [reply core/src/main/scala/io/chrisdavenport/monoids/Dual.scala:6]: Should seriously consider writing these as opaque types in dotty, and leave this in scala 2. The performance hit if we box here can be pretty brutal.

File: build.sbt (hunk 1 above)
Comment (ChristopherDavenport) [reply build.sbt:44]: This is correct, but we need the files to move to that package structure as well.

**Comment by danicheg** [comment]
I've opened an issue for discussing migration on Scala 3 #219 due to a bug with `AnyVal`is currently not fixed in Scala 3.

**Comment by danicheg** [comment]
Thanks for that work on Scala 3 migration. Some ideas conversed here was used in #225.

**Pull request closed without merging**